client/src/main/java/io/pravega/client/control/impl/ControllerImpl.java

```diff
@@ -1580,6 +1581,30 @@ private void closeChannel() {
         }
     }
 
+    @Override
+    public CompletableFuture<KeyValueTableConfiguration> getKeyValueTableConfiguration(String scope, String kvtName) {
+        Exceptions.checkNotClosed(closed.get(), this);
+        final long requestId = requestIdGenerator.get();
+        long traceId = LoggerHelpers.traceEnter(log, GET_KEY_VALUE_TABLE_CONFIGURATION, scope, kvtName, requestId);
+        String scopedKvtName = NameUtils.getScopedKeyValueTableName(scope, kvtName);
+
+        final CompletableFuture<KeyValueTableConfig> result = this.retryConfig.runAsync(() -> {
+            RPCAsyncCallback<KeyValueTableConfig> callback = new RPCAsyncCallback<>(requestId,
+                    GET_KEY_VALUE_TABLE_CONFIGURATION, scope, kvtName);
+            new ControllerClientTagger(client, timeoutMillis).withTag(requestId, GET_KEY_VALUE_TABLE_CONFIGURATION,
+                    scope, kvtName)
+                    .getKeyValueTableConfiguration(ModelHelper.createKeyValueTableInfo(scope, kvtName), callback);
+            return callback.getFuture();
+        }, this.executor);
+        return result.thenApplyAsync(ModelHelper::encode, this.executor)
```

Comment:
Can we confirm what the behavior will be if the KVT does not exist? Will it:
1. Throw an exception (i.e., return a failed future with an exception)? If so, which one?
2. Return null?

Let's update this method's contract in `Controller.java` to clearly define the behavior in this case.

Reply:
I've added status to the GRPC server interaction. The server now returns a status indicating if the getting the config for the KVT was successful or not. Including a status for in case the KVT in question is not found.